pkg/lvgl/contrib/lvgl.c

```diff
@@ -80,34 +89,72 @@ static void *_task_thread(void *arg)
 
 static void _disp_map(lv_disp_drv_t *drv, const lv_area_t *area, lv_color_t *color_p)
 {
-    if (!_dev) {
+    if (!_screen_dev->display) {
         return;
     }
 
-    disp_dev_map(_dev, area->x1, area->x2, area->y1, area->y2,
+    disp_dev_map(_screen_dev->display, area->x1, area->x2, area->y1, area->y2,
                  (const uint16_t *)color_p);
 
     LOG_DEBUG("[lvgl] flush display\n");
 
     lv_disp_flush_ready(drv);
 }
 
-void lvgl_init(disp_dev_t *dev)
+#ifdef MODULE_TOUCH_DEV
+static bool _touch_read(lv_indev_drv_t *indev_driver, lv_indev_data_t *data)
+{
+    if (!_screen_dev->touch) {
+        return false;
+    }
+
+    (void)indev_driver;
+    static lv_coord_t last_x = 0;
+    static lv_coord_t last_y = 0;
+
+    touch_t positions[1];
+    uint8_t touches = touch_dev_touches(_screen_dev->touch, positions, 1);
+    data->state = (touches > 0) ? LV_INDEV_STATE_PR : LV_INDEV_STATE_REL;
+
+    if (data->state == LV_INDEV_STATE_PR) {
+        last_x = positions[0].x;
+        last_y = positions[0].y;
+    }
```

Comment:
So we only get the coordinates on touch, not on release?
Better add a comment. :)

(What about press and drag?)

Reply:
This is how lvgl handles a touch device. For press and drag, you would get all positions while pressed and need some logic to manage the release event.

Reply:
I'd still add a comment. It's not immediately obvious what `LV_INDEV_STATE_PR` is or why we need to cache old touch positions.

Reply:
Done. I just copy pasted the comment from the original usage documentation: https://github.com/lvgl/lvgl/tree/v6.1.2#add-littlevgl-to-your-project